### PR TITLE
Clarify documentation of clock_recovery_mm_xx

### PR DESCRIPTION
**Pull request by ThomasHabets**
Information was taken from
https://www.tablix.org/~avian/blog/archives/2015/03/notes_on_m_m_clock_recovery/
and would be useful to have available for beginners.

### CODE REVIEW

**Comment by ThomasHabets** [comment]
If this is not the right place for this kind of documentation, then please point me to where it should be.

File: gr-digital/include/gnuradio/digital/clock_recovery_mm_cc.h [hunk 1]
```diff
@@ -37,6 +37,18 @@ namespace gr {
      * This implements the Mueller and Müller (M&M) discrete-time
      * error-tracking synchronizer.
      *
+     * The input signal must have values around zero, as it works with
+     * sign changes of its input.
+     *
```
Comment (awalls-cx18) [inline gr-digital/include/gnuradio/digital/clock_recovery_mm_cc.h]: "The peak to peak input signal amplitude must be symmetrical about zero, as the M&M timing error detector (TED) is a decision directed TED, and this block uses a symbol decision slicer referenced at zero.

The input signal peak amplitude should be controlled to a consistent level (e.g. +/- 1.0) before this block to achieve consistent results for given gain settings; as the TED's output error signal is directly affected by the input amplitude.

File: gr-digital/include/gnuradio/digital/clock_recovery_mm_cc.h [hunk 2]
```diff
@@ -37,6 +37,18 @@ namespace gr {
...
+     * Input can not be a square wave. Use blocks_moving_average_xx
+     * over samples per symbol to turn square wave into suitable input
+     * to this block.
+     *
```
Comment (awalls-cx18) [inline gr-digital/include/gnuradio/digital/clock_recovery_mm_cc.h]: "The input signal must have peaks in order for the TED to output a correct error signal.  If the input signal pulses do not have peaks (e.g. rectangular pulses) the input signal should be conditioned with a matched pulse filter or other appropriate filter to peak the input pulses.  For a rectangular base pulse that is N samples wide, the matched filter taps would be [1.0/float(N)]*N"

**Comment by ThomasHabets** [comment]
I've used your suggested replacements with a slight clarification for beginners not yet read up on what taps are.

Hopefully I didn't in doing so make it incorrect.

**Comment by awalls-cx18** [comment]
Looks OK to me.  I have a hard time writing for beginners, so hopefully, my suggetions weren't to opaque or offputting for a novice.

**Comment by jmcorgan** [comment]
First, thanks for doing this.

As a contributor, there are a few practices we ask people to follow to help keep our code tree and commits organized.  Since this is a bug fix, the normal practice is to:
- Create a bug fix branch off maint, call it something descriptive related to the bug
- Request the branch to be merged to maint in the pull request
- Combine related/overlapping commits into a single one
- Use our commit style "module: fixed..." where module in this case would be "digital".

This isn't intended as a criticism as you are new (and again, we appreciate this!), but it is help to get into these habits early.

For this PR I will go ahead and clean things up, so you don't have to do anything, but please keep the above in mind for the future.  Welcome to the project.

**Comment by ThomasHabets** [comment]
I'll keep it in mind for next time, but no action required for me this time, right?

**Comment by jmcorgan** [comment]
Right.

**Comment by jmcorgan** [comment]
These two commits have been combined into one and merged onto the maint branch.

**Pull request closed without merging**